src/generate/generic.rs

```diff
@@ -0,0 +1,87 @@
+use quote::Tokens;
+
+use crate::errors::*;
+
+/// Generates generic bit munging code
+pub fn render() -> Result<Vec<Tokens>> {
+    let mut code = vec![];
+    let mut generic_items = vec![];
+
+    generic_items.push(quote! {
+        use core::marker;
+
+        ///Converting enumerated values to bits
+        pub trait ToBits<N> {
+            ///Convertion method
+            fn _bits(&self) -> N;
+        }
+    });
+
+    generic_items.push(quote! {
+        ///Value read from the register
+        pub struct FR<U, T> {
```

Comment:
Why use two types here? The code below always generates the same for both and just for `PhantomType` it shouldn't matter anyway.

Reply:
Second type is used for enumerated values if they present and it's also needed for differ one reader from another.
If enumerated values are absent I just ignore second type.

Reply:
> it's also needed for differ one reader from another

Huh?

> If enumerated values are absent I just ignore second type.

I noticed which makes me wonder what the purpose is. Where can I see an example of such an enumerated type?

Reply:
https://docs.rs/stm32f1/0.7.1/stm32f1/stm32f103/gpioa/crh/enum.CNF8R.html#method.is_push_pull

Aside from `bits` method reader have also `is_push_pull` and other methods generated from `<enumeratedValues>` list.

Reply:
Same with writers but more complex.